spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -93,10 +93,10 @@ protected CloseableIterator<ColumnarBatch> open(FileScanTask task) {
     InputFile inputFile = getInputFile(filePath);
     Preconditions.checkNotNull(inputFile, "Could not find InputFile associated with FileScanTask");
 
-    SparkDeleteFilter deleteFilter =
-        task.deletes().isEmpty()
-            ? null
-            : new SparkDeleteFilter(filePath, task.deletes(), counter());
+    SparkDeleteFilter sparkDeleteFilter =
+        new SparkDeleteFilter(filePath, task.deletes(), counter(), true);
+
+    SparkDeleteFilter deleteFilter = task.deletes().isEmpty() ? null : sparkDeleteFilter;
```

Comment:
We don't need delete filter if `task.deletes().isEmpty()`, but the new code always creates a filter object.
```
filter = null;
if(!task.deletes().isEmpty()) {
 filter = new filter()
}
```